code-challenges.test.js

```diff
@@ -0,0 +1,144 @@
+                
+                expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+                expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+                expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+            })
+        })
+
+// b) Create the function that makes the test pass.
+const codedMessage = (str) =>{
+    const vowels = {
+      'a': '4',
+      'e': '3',
+      'i': '1',
+      'o': '0'
+    };
+  
+    let codedString =""
+    for (let i = 0; i < str.length; i++) {
+      let char = str[i].toLowerCase();
+      if (vowels.hasOwnProperty(char)) {
+        codedString += vowels[char];
+      } else {
+        codedString += char;
+      }
+    }
+  
+    return codedString;
+  }
+  
+
+// Pseudo code:
+// create a  function named codedMessages that will take a string
+// inside the funtion create an object with the vowels as key and the return number as value
+// create a variable namedcodedString that will hold an empty string as a placeholder for the string being return.
+//create a for loop that will iterate throught the passed in string
+// use and if else statement to replace each vowel with the corresponding number assigned in the vowels obeject vowels by checking if the current index  lowercased matches a key in the vowels object replace it.
+// return the new string
+
+
+// --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
+
+// a) Create a test with expects statement using the variable provided.
+        describe("wordsWithLetter", () => {
+                it("takes in an array of words and a single letter and returns an array of all the words containing that particular letter.", () => {
+
+                const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+                const filterLetterA = "a"
+                const filterLetterE = "e"
+                
+                expect(wordsWithLetter(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+                expect(wordsWithLetter(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+            })
+        })
+
+// b) Create the function that makes the test pass.
+const wordsWithLetter = (words, letter) => {
+    const wordsWithLetter = [];
+    
+    for (let i = 0; i < words.length; i++) {
+      const word = words[i];
+      
+      if (word.includes(letter)) {
+        wordsWithLetter.push(word);
+      }
+    }
+    
+    return wordsWithLetter;
+  }
+
+// Pseudo code:
+// create function named wordsWithLetter it takes two arguments and array and a letter
+// inside the function create an array to hold the return
+// create a for loop to iterate throught the array 
+// create and if statement that will push the word into the new array if it contains the given letter
+// return the new array
```

Comment:
Think about what HOF does all of this logic for you:
returns an array, has a built-in conditional statement.